sympy/sets/sets.py

```diff
@@ -1202,28 +1202,32 @@ def reduce(args):
 
         # ===== Pair-wise Rules =====
         # Here we depend on rules built into the constituent sets
-        args = set(args)
+        args = FiniteSet(*args)
```

Comment:
I am trying to use only `frozenset` and `ordered` in [line 1205](https://github.com/sympy/sympy/pull/11188/files#diff-7b3e9b0852f1b1d094f87233f56e201fR1205), but still error in coming (each time different assertion error for different python version).

```
args = list(ordered(frozenset(tuple(args)), Set._infimum_key))
# or 
args = set(frozenset(args))
```